### PR TITLE
Update Follow button UI

### PR DESCRIPTION
**Pull request by yaelirub**
Following design feedback in: #14612 

In this PR:
- Added 2 colors, buttonIcon and primaryButtonBorder to UIColor+MurielColors.
- updated button font from headline to callout
- updated following font from semibold to regular
- updated button's border color to opaqueSeparator
- updated following background color to clear instead of black
- update following text and icon color to .systemGray2

#### Dark

Follow             |  Following 
:-------------------------:|:-------------------------:
  |  

#### Light

Follow             |  Following 
:-------------------------:|:-------------------------:
  |  

To test:
1. Go to reader
2. Tap on a site name in a post card
3. See the follow / following button looks good and applies designs updates 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift [hunk 1]
```diff
@@ -208,6 +208,14 @@ extension UIColor {
         return muriel(color: .divider)
     }
 
+    static var primaryButtonBorder: UIColor {
+           if #available(iOS 13, *) {
+               return .opaqueSeparator
+           }
+
+        return UIColor(displayP3Red: 198.0, green: 198.0, blue: 200.0, alpha: 1.0)
```
Comment (leandroalonso) [inline WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift:216]: Wouldn't that be the same as `muriel(color: .gray, .shade10)`?

File: WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift [hunk 2]
```diff
@@ -267,6 +275,14 @@ extension UIColor {
         return UIColor.neutral(.shade20)
     }
 
+    static var buttonIcon: UIColor {
+        if #available(iOS 13, *) {
+            return .systemGray2
+        }
+
+        return UIColor.neutral(.shade5)
```
Comment (leandroalonso) [inline WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift:283]:
```suggestion
        return .neutral(.shade5)
```

**Comment by mbshakti** [comment]
Good point @leandroalonso. Yael can we use `label secondary` for the button copy instead of `systemGray2`? Apologies for the back and forth on this, thank you!

File: WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift (hunk 1 above)
Comment (yaelirub) [reply WordPress/Classes/Extensions/Colors and Styles/UIColor+MurielColors.swift]: updated in https://github.com/wordpress-mobile/WordPress-iOS/pull/14621/commits/9fc13a87221b9e690eecaa21b7d6018111b2613d

**Comment by yaelirub** [comment]
@leandroalonso , ready for another look

**Pull request merged**